tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp

```diff
@@ -244,50 +244,49 @@ void test_inverse_map(const Map& map,
  */
 template <typename Map>
-void test_map3d(const Map& map) {
+void test_suite_for_map(const Map& map) {
   // Set up random number generator
   std::random_device rd;
+  const auto seed = rd();
   std::mt19937 gen(rd());
+  INFO("seed = " << seed);
   std::uniform_real_distribution<> real_dis(-1.0, 1.0);
-  const double xi = real_dis(gen);
-  CAPTURE_PRECISE(xi);
-  const double eta = real_dis(gen);
-  CAPTURE_PRECISE(eta);
-  const double zeta = real_dis(gen);
-  CAPTURE_PRECISE(zeta);
 
-  const std::array<std::array<double, 3>, 13> test_points{{{{0.0, 0.0, 0.0}},
-                                                          {{-1.0, -1.0, -1.0}},
-                                                          {{1.0, -1.0, -1.0}},
-                                                          {{-1.0, 1.0, -1.0}},
-                                                          {{1.0, 1.0, -1.0}},
-                                                          {{-1.0, -1.0, 1.0}},
-                                                          {{1.0, -1.0, 1.0}},
-                                                          {{-1.0, 1.0, 1.0}},
-                                                          {{1.0, 1.0, 1.0}},
-                                                          {{-0.1, 0.3, 0.1}},
-                                                          {{0.5, 0.7, -0.5}},
-                                                          {{0.9, -1.0, 0.4}},
-                                                          {{xi, eta, zeta}}}};
-  test_serialization(map);
-  CHECK_FALSE(map != map);
-  test_coordinate_map_argument_types(map, test_points[0]);
-  for (const auto& point : test_points) {
-    test_jacobian(map, point);
-    test_inv_jacobian(map, point);
-    test_inverse_map(map, point);
+  std::array<double, Map::dim> origin{};
+  std::array<double, Map::dim> random_point{};
+  for (size_t i = 0; i < Map::dim; i++) {
+    gsl::at(origin, i) = 0.0;
+    gsl::at(random_point, i) = real_dis(gen);
   }
+
+  const auto test_helper =
+      [&origin, &random_point ](const auto& lmap) noexcept {
+    test_serialization(lmap);
+    CHECK_FALSE(lmap != lmap);
+    test_coordinate_map_argument_types(lmap, origin);
+
+    test_jacobian(lmap, origin);
+    test_inv_jacobian(lmap, origin);
+    test_inverse_map(lmap, origin);
+
+    for (VolumeCornerIterator<Map::dim> vci{}; vci; ++vci) {
+      test_jacobian(lmap, vci.coords_of_corner());
+      test_inv_jacobian(lmap, vci.coords_of_corner());
+      test_inverse_map(lmap, vci.coords_of_corner());
+    }
+
+    test_jacobian(lmap, random_point);
+    test_inv_jacobian(lmap, random_point);
+    test_inverse_map(lmap, random_point);
+  };
+  test_helper(map);
   const auto map2 = serialize_and_deserialize(map);
   CHECK(map2 == map);
-  for (const auto& point : test_points) {
-    test_jacobian(map2, point);
-    test_inv_jacobian(map2, point);
-    test_inverse_map(map2, point);
-  }
+  test_helper(map2);
```

Comment:
CHECK(map2(random_point) == map(random_point) or an existing function that does the same thing but doesn't do it by calling operator== or != . try check_if_maps_are_equal